### ISSUE TITLE
View Assignment: LayoutStride -> LayoutLeft and LayoutStride -> LayoutRight

### ISSUE THREAD

**Issue by crtrott** [kokkos/kokkos#1594]
Make this assignable with runtime error checking. @AndrewGaspar

**Comment by AndrewGaspar** [comment]
It would also be nice to have built-in routines for checking if a LayoutStride is compatible with LayoutRight so you don't have to check the strides yourself.

**Comment by crtrott** [comment]
Something like this? 
```c++
template<class ViewTypeA, class ViewTypeB>
bool is_assignable(const ViewTypeA& a, const VIewTypeB& b);
```

**Comment by crtrott** [comment]
Btw. if you feel you have time to do something about this let me know ;-). 
The place for this is basically found in:
core/src/impl/Kokkos_ViewMapping.hpp

**Comment by AndrewGaspar** [comment]
Yeah, I think I more or less understand the code. I may be able to come up with a fix. The open policy here at LANL is weird, so it may be easier if I just do it on my own time. But it would probably be useful, anyway, to just submit the approval request to contribute to Kokkos.

**Comment by crtrott** [comment]
It might be worthwhile for us to figure out what the policy is with respect to contributing to Kokkos etc. from LANL. From our side we know what all the rules are (for example around auto testing etc.) and we can basically treat you guys as if you were at Sandia. I might talk to Galen and see if we can get guidance on this.
